Automattic/wp-calypso
# E2E: add support to interact with editors with Gutenberg 14.9.1 inner iframes.

#### Proposed Changes

This PR adds support for interacting with editors that have the inner iframes, introduced with Gutenberg 14.9.1.

Previous PR: https://github.com/Automattic/wp-calypso/pull/72112
Context: p1673294397010529-slack-CBTN58FTJ and p1673397395255099-slack-C1A1EKDGQ

In short:
Gutenberg 14.9.1 introduces a nested iframe at the editor canvas level if a block-based theme is used. 
If an iframe wraps the editor canvas the existing POMs are no longer able to interact with editor elements (eg. blocks).

In this PR, by using a new optional parameter to tell the EditorPage about the usage of block-based themes we can add support for both non-block-based and block-based themes. 

Key changes:
- add optional parameter `blockTheme` which defaults to `false`.
- add new class property `editorCanvas` that refers to jus the editor canvas. 

#### Testing Instructions

Ensure the following build configurations are passing:
  - [x] i18n E2E

Additionally, change the theme for a user used in the Gutenberg suite of tests to a block-based theme and ensure the spec continues to pass.

#### Pre-merge Checklist





Related to #
